cdc/scheduler/internal/tp/scheduler_move_table.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tp
+
+import (
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/cdc/model"
+	"go.uber.org/zap"
+)
+
+var _ scheduler = &moveTableScheduler{}
+
+type moveTableScheduler struct {
+	tasks map[model.TableID]*scheduleTask
+}
+
+func newMoveTableScheduler() *moveTableScheduler {
+	return &moveTableScheduler{
+		tasks: make(map[model.TableID]*scheduleTask),
+	}
+}
+
+func (m *moveTableScheduler) Name() string {
+	return string(schedulerTypeMoveTable)
+}
+
+func (m *moveTableScheduler) addTask(tableID model.TableID, target model.CaptureID) bool {
+	// previous triggered task not received yet, decline the new manual move table request.
+	if _, ok := m.tasks[tableID]; ok {
+		return false
+	}
+	m.tasks[tableID] = &scheduleTask{
+		moveTable: &moveTable{
+			TableID:     tableID,
+			DestCapture: target,
+		},
+		accept: func() {
+			delete(m.tasks, tableID)
```

Comment:
Could you make it thread-safe? Even though it's single threaded now, I'd like to make it decoupled with other components implementation.